iocore/hostdb/P_RefCountCache.h

```diff
@@ -169,7 +170,7 @@ template <class C> class RefCountCachePartition
 
   hash_type &get_map();
 
-  Ptr<ProxyMutex> lock; // Lock
+  ink_rwlock lock = PTHREAD_RWLOCK_WRITER_NONRECURSIVE_INITIALIZER_NP;
```

Comment:
Pretty interesting change. Could you use `std::shared_mutex` here? 
We can do it later, but using `std::shared_mutex` will make it easier to apply `ts::bravo::shared_mutex` or other reader-writer lock improvements in the future, IMO.

Reply:
Maybe I need to since apparently freebsd/osx don't support PTHREAD_RWLOCK_WRITER_NONRECURSIVE_INITIALIZER_NP. :)